hack/verify-staticcheck.sh

```diff
@@ -13,32 +13,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# - download golangci-lint if needed (replace existing if incorrect version)
+# - run golangci-lint
 
 set -o errexit
 set -o nounset
 set -o pipefail
 
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
-GOLANGCI_LINT_VER="v1.59.0"
+GOLANGCI_LINT_VER="1.59.0"
+GOLANGCI_LINT="./bin/golangci-lint"
```

Comment:
seems like you want to download the tools to the project folder and not `GOPATH`, I think this makes sense, but currently karmada downloads tools into `GOPATH`, which might be worth discussing.

Reply:
yeah, other tools would need to follow that pattern, so better get consensus
idk of any reason not to do it 🤷

Reply:
I don't have much background here, I think it would be possible to add a parameter to have the tool download to the specifically directory, defaulting to gopath, what do you think?  

Maybe:
```shell
GO_PATH=$(go env GOPATH)
TOOLS_PATH=${TOOLS_PATH:-GO_PATH}
```

Reply:
cc @zhzhuang-zju @XiShanYongYe-Chang @RainbowMango

Reply:
that would fix the error

if we also want the correct version it would need to switch and replace the binaries in gopath depending on what version the current project needs, should mostly be fine just weird when using different projects with different version requirements

Reply:
I think we can provide the capability to install to a specific path, but considering that minimizing the impact on users who may have already installed their own `golang-lint`, the default path could be set to `./bin/`